relay-sampling/src/config.rs

```diff
@@ -422,20 +462,9 @@ mod tests {
     }
 
     #[test]
-    fn test_sampling_config_with_rules_and_rules_v2_deserialization() {
+    fn test_legacy_deserialization() {
         let serialized_rule = r#"{
-               "rules": [
-                  {
-                     "sampleRate": 0.5,
-                     "type": "trace",
-                     "active": true,
-                     "condition": {
-                        "op": "and",
-                        "inner": []
-                     },
-                     "id": 1000
-                 }
-               ],
+               "rules": [],
```

Comment:
Legacy actually required serializing an empty `rules` array since in even earlier versions of Relay, that `skip_deserializing` annotation was missing. I'll keep it here to reflect the actual payload that's sent by these versions. 

This is important, because without these empty rules we could've just used a `serde(alias)`.

Reply:
Thanks, I confused `skip_deserializing` with `skip_serializing` here.